core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -1196,7 +1205,9 @@ private boolean bindAsUser(String userName, String dn, Object credentials) throw
                 log.debug("Authentication failed " + e);
                 log.debug("Clearing cache for DN: " + dn);
             }
-            userCache.remove(userName);
+            if (userName != null) {
+                userCache.remove(userName);
+            }
```

Comment:
This null check is done only to avoid the NPE in userCache ConcurrentHashMap. If the username is null `userCache.remove(userName)` will throw an NPE.